maggy/core/environment/databricks.py

```diff
@@ -0,0 +1,52 @@
+#
+#   Copyright 2021 Logical Clocks AB
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+import os
+import shutil
+
+from maggy import util
+from maggy.core.environment.base import BaseEnv
+
+
+class DatabricksEnv(BaseEnv):
+    """
+    Environment implemented for maggy usage on Databricks.
+    """
+
+    def __init__(self):
+        self.log_dir = "/dbfs/maggy_log/"
+        if not os.path.exists(self.log_dir):
+            os.mkdir(self.log_dir)
+
+    def mkdir(self, hdfs_path):
+        return os.mkdir(hdfs_path)
+
+    def project_path(self, project=None, exclude_nn_addr=False):
+        return "/dbfs/"
+
+    def get_executors(self, sc):
+        try:
+            if sc._conf.get("spark.databricks.clusterUsageTags.clusterScalingType") == "autoscaling":
+                maxExecutors = int(sc._conf.get("spark.databricks.clusterUsageTags.clusterMaxWorkers"))
+            else:
+                maxExecutors = int(sc._conf.get("spark.databricks.clusterUsageTags.clusterWorkers"))
+
+            return maxExecutors
+        except:  # noqa: E722
+            raise RuntimeError(
+                "Failed to find some of the spark.databricks properties."
+            )
```

Comment:
The error won't tell the user what's going on, can you split this into two try/except clauses, each trying to access the different properties and then in the error message you can be specific on which one it couldn't find.

Reply:
ah and try to catch the exact exception, not just any, it probably throws a KeyError or the like